pkg/clusteragent/admission/metrics/metrics.go

```diff
@@ -60,10 +60,10 @@ var (
 		telemetry.Options{NoDoubleUnderscoreSep: true},
 	)
 	LibInjectionAttempts = telemetry.NewCounterWithOpts("admission_webhooks", "library_injection_attempts",
-		[]string{"language", "injected", "auto_detected"}, "Number of pod library injection attempts by language.",
+		[]string{"language", "injected", "auto_detected", "injection_type", "service_name"}, "Number of pod library injection attempts by language.",
```

Comment:
by adding `service_name` you are increasing a lot the metric cardinality. 
have we evaluated the cost of adding it?

Reply:
That was my concern as well. There are [some cases](https://metabase-analytics.us1.prod.dog/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7ImRhdGFiYXNlIjo0MywidHlwZSI6InF1ZXJ5IiwicXVlcnkiOnsic291cmNlLXRhYmxlIjo2NDU1MywiZmlsdGVyIjpbImFuZCIsWyI9IixbImZpZWxkIiwxMjM4NDE0LG51bGxdLCJkYXRhZG9nLmNsdXN0ZXJfYWdlbnQuYWRtaXNzaW9uX3dlYmhvb2tzLmxpYnJhcnlfaW5qZWN0aW9uX2F0dGVtcHRzIl0sWyJub3QtZW1wdHkiLFsiZmllbGQiLDEyMzg0MDksbnVsbF1dXX19LCJkaXNwbGF5IjoidGFibGUiLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7fX0=) when service is already set, but I'm not sure where the service tag is coming from.

Reply:
To reduce the risk of exploding cardinality, it might be better to remove the service tag here and rely on library startup events emitted by client libraries to get injection stats per service.

Reply:
the service might be the service of the cluster-agent pod.

Reply:
Yea, this makes sense given relatively [low cardinality](https://metabase-analytics.us1.prod.dog/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7ImRhdGFiYXNlIjo0MywidHlwZSI6InF1ZXJ5IiwicXVlcnkiOnsic291cmNlLXRhYmxlIjo2NDU1MywiYWdncmVnYXRpb24iOltbImNvdW50Il1dLCJicmVha291dCI6W1siZmllbGQiLDEyMzg0MDksbnVsbF1dLCJvcmRlci1ieSI6W1siZGVzYyIsWyJhZ2dyZWdhdGlvbiIsMF1dXSwiZmlsdGVyIjpbImFuZCIsWyI9IixbImZpZWxkIiwxMjM4NDE0LG51bGxdLCJkYXRhZG9nLmNsdXN0ZXJfYWdlbnQuYWRtaXNzaW9uX3dlYmhvb2tzLmxpYnJhcnlfaW5qZWN0aW9uX2F0dGVtcHRzIl0sWyJub3QtZW1wdHkiLFsiZmllbGQiLDEyMzg0MDksbnVsbF1dXX19LCJkaXNwbGF5IjoiYmFyIiwiZGlzcGxheUlzTG9ja2VkIjp0cnVlLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7fSwib3JpZ2luYWxfY2FyZF9pZCI6bnVsbH0=) of the services